cascading/src/main/java/org/elasticsearch/hadoop/cascading/EsHadoopScheme.java

```diff
@@ -83,12 +80,13 @@
     public void sourcePrepare(FlowProcess<JobConf> flowProcess, SourceCall<Object[], RecordReader> sourceCall) throws IOException {
         super.sourcePrepare(flowProcess, sourceCall);
 
-        Object[] context = new Object[3];
+        Object[] context = new Object[4];
         context[0] = sourceCall.getInput().createKey();
         context[1] = sourceCall.getInput().createValue();
         // as the tuple _might_ vary (some objects might be missing), we use a map rather then a collection
         Settings settings = loadSettings(flowProcess.getConfigCopy(), true);
         context[2] = CascadingUtils.alias(settings);
+        context[3] = settings.getOutputAsJson();
```

Comment:
Would it make sense to use a constant for the index in the context array, something like 
` context[OUTPUT_AS_JSON] = settings.getOutputAsJson();`

That would help to read the context array as well as quickly identify where it is read & write.